lib/pulsar/producer.ex

```diff
@@ -101,6 +113,9 @@ defmodule Pulsar.Producer do
     {compression, genserver_opts} = Keyword.pop(genserver_opts, :compression, :NONE)
     {chunking_enabled, genserver_opts} = Keyword.pop(genserver_opts, :chunking_enabled, false)
     {max_message_size, genserver_opts} = Keyword.pop(genserver_opts, :max_message_size, 5_242_880)
+    {batch_enabled, genserver_opts} = Keyword.pop(genserver_opts, :batch_enabled, false)
```

Comment:
@giusdp - I am wondering if we need to take `chunking_enabled` into account. As per Pulsar's official documentation, `chunking` and `batching` cannot be enabled at the same time (but I haven't tested things myself 😄). No need to address this now, we could address this in the future. The (remote) broker may take care of this by responding with an error if we try to send a batched message when `chunking` is enabled. If you are curious to know, we could add a simple test for this. Otherwise, let's tackle this in the future.